src/snowflake/snowpark/modin/plugin/compiler/snowflake_query_compiler.py

```diff
@@ -15494,10 +15538,46 @@ def dt_floor(
                 - 'raise' will raise an NonExistentTimeError if there are nonexistent times.
         Returns:
             A new QueryCompiler with floor values.
-
         """
-        ErrorMessage.not_implemented(
-            "Snowpark pandas doesn't yet support the method 'Series.dt.floor'"
+        if ambiguous != "raise":
+            ErrorMessage.not_implemented(
+                "Snowpark pandas 'Series.dt.floor' method doesn't yet support 'ambiguous' parameter"
+            )
+        if nonexistent != "raise":
+            ErrorMessage.not_implemented(
+                "Snowpark pandas 'Series.dt.floor' method doesn't yet support 'nonexistent' parameter"
+            )
+        internal_frame = self._modin_frame
+
+        slice_length, slice_unit = rule_to_snowflake_width_and_slice_unit(
+            rule=freq  # type:  ignore[arg-type]
+        )
+
+        if slice_unit not in SUPPORTED_DT_FLOOR_CEIL_FREQS:
+            ErrorMessage.not_implemented(
+                "Snowpark pandas 'Series.dt.floor' method doesn't support setting 'freq' parameter to '{freq}'"
+            )
```

Comment:
This chunk of code seems duplicated, can we just move this to a utils function checking whether the operation is supported -- the only difference appears to be `floor` vs. `ceil` in the error mesages?

Reply:
Well, it does more than just checking whether the operation is supported. It also computes the slice length and slice unit, which is used later in the code in slightly different ways for floor and ceil.